pkg/controller/health_check_test.go

```diff
@@ -76,80 +72,54 @@ func (s *TestHealthCheckSuite) TestHealthCheckHandler() {
-
-		// Check the response body is what we expect.
-		data := &controller.Health{}
-		err := json.Unmarshal(rr.Body.Bytes(), &data)
-		require.NoError(s.T(), err)
+	s.Run("service Unavailable", func() {
+		// Setting production mode
+		s.Config.GetViperInstance().Set("testingmode", false)
 
-		val := data.Revision
-		assert.Equal(s.T(), configuration.Commit, val, "wrong revision in health response, got %s want %s", val, configuration.Commit)
-	})
+		healthCheckCtrl := controller.NewHealthCheck(logger, s.Config, &mockHealthChecker{})
+		handler := gin.HandlerFunc(healthCheckCtrl.GetHandler)
 
-	s.Run("build time", func() {
 		// We create a ResponseRecorder (which satisfies http.ResponseWriter) to record the response.
 		rr := httptest.NewRecorder()
 		ctx, _ := gin.CreateTestContext(rr)
 		ctx.Request = req
 
-		// Our handlers satisfy http.Handler, so we can call their ServeHTTP method
-		// directly and pass in our Request and ResponseRecorder.
 		handler(ctx)
 
 		// Check the status code is what we expect.
-		assert.Equal(s.T(), rr.Code, http.StatusOK, "handler returned wrong status code: got %v want %v", rr.Code, http.StatusOK)
+		assert.Equal(s.T(), http.StatusServiceUnavailable, rr.Code, "handler returned wrong status code")
 
 		// Check the response body is what we expect.
 		data := &controller.Health{}
 		err := json.Unmarshal(rr.Body.Bytes(), &data)
 		require.NoError(s.T(), err)
 
-		val := data.BuildTime
-		assert.Equal(s.T(), configuration.BuildTime, val, "wrong build_time in health response, got %s want %s", val, configuration.BuildTime)
+		assertHealth(s.T(), false, false, data)
 	})
+}
 
-	s.Run("start time", func() {
-		// We create a ResponseRecorder (which satisfies http.ResponseWriter) to record the response.
-		rr := httptest.NewRecorder()
-		ctx, _ := gin.CreateTestContext(rr)
-		ctx.Request = req
-
-		// Our handlers satisfy http.Handler, so we can call their ServeHTTP method
-		// directly and pass in our Request and ResponseRecorder.
-		handler(ctx)
-
-		// Check the status code is what we expect.
-		assert.Equal(s.T(), rr.Code, http.StatusOK, "handler returned wrong status code: got %v want %v", rr.Code, http.StatusOK)
+func assertHealth(t *testing.T, expectedAlive, expectedTestingMode bool, actual *controller.Health) {
+	assert.Equal(t, expectedAlive, actual.Alive, "wrong alive in health response")
+	assert.Equal(t, configuration.Commit, actual.Revision, "wrong revision in health response")
+	assert.Equal(t, configuration.BuildTime, actual.BuildTime, "wrong build_time in health response")
+	assert.Equal(t, configuration.StartTime, actual.StartTime, "wrong start_time in health response")
+	assert.Equal(t, expectedTestingMode, actual.TestingMode, "wrong testing mode in health response")
+}
 
-		// Check the response body is what we expect.
-		data := &controller.Health{}
-		err := json.Unmarshal(rr.Body.Bytes(), &data)
-		require.NoError(s.T(), err)
+type mockHealthChecker struct {
```

Comment:
what about using minimock to generate the mock impl since we have interfaces?

Reply:
To be honest I'm not a fun of minimock :) IMO sometimes it adds additional complexity instead of reducing it. It requires build generation.
Anyway in this particular case I think it would be overkill. The mock checker I'm providing here is super light.

Reply:
ok, I understand your concern. I believe it becomes helpful when you need to alter the behaviour of the mock in the tests to verify different uses cases. In which case, having multiple implementations as you're doing here would be a less elegant approach. But we can keep it as-is for now.

Reply:
What multiple implications would we need here? Wouldn’t the current mock implementation cover all possible needs (as it does in these tests)?

Reply:
ok, this mock is simple so we're fine. But I believe that minimock (for example) provides more flexibility if we need to alter the response to a method call to test so cases of errors, etc. Here, it's just a bool that is returned, and it is provided when initializing the mock, so it's good enough, indeed.